src/contracts/permissions/PermissionController.sol

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.27;
+
+import "@openzeppelin-upgrades/contracts/proxy/utils/Initializable.sol";
+import "./PermissionControllerStorage.sol";
+
+contract PermissionController is Initializable, PermissionControllerStorage {
+    using EnumerableSet for *;
+
+    modifier onlyAdmin(
+        address account
+    ) {
+        require(isAdmin(account, msg.sender), NotAdmin());
+        _;
+    }
+
+    /**
+     *
+     *                         INITIALIZING FUNCTIONS
+     *
+     */
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize() external initializer {}
+
+    /**
+     *
+     *                         EXTERNAL FUNCTIONS
+     *
+     */
+
+    /// @inheritdoc IPermissionController
+    function setAdmin(address account, address admin) external onlyAdmin(account) {
+        // Add the admin to the account's admins
+        // If the admin is already set, the set will fail
+        EnumerableSet.AddressSet storage admins = _permissions[account].admins;
+        require(admins.add(admin), AdminAlreadySet());
+
+        emit AdminSet(account, admin);
+    }
+
+    function removeAdmin(address account, address admin) external onlyAdmin(account) {
```

Comment:
So if you remove all admins, the original account becomes the admin again?
Is this desired? Or can we just prevent ever having 0 admins when you try to remove.

Reply:
Or as Michael said prevent having 0 when you remove

Reply:
Let's prevent 0. We want to specifically enable permanent key rotation

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/870/commits/eac8bbe1116ca3cef86eaada1f9d6d666aff49f7